### PR TITLE
simplify several classes

### PR DESCRIPTION
**Pull request by rotty3000**
Signed-off-by: Raymond Augé <raymond.auge@liferay.com>

### CODE REVIEW

File: org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceConfiguration.java [hunk 1]
```diff
@@ -17,18 +17,28 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.function.Function;
+
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.Filter;
+import org.osgi.framework.ServiceReference;
 import org.osgi.test.common.tracking.TrackServices;
```
Comment (bjhargrave) [inline org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceConfiguration.java:32]: Is there still a good reason to leave TrackServices in its own package? At the minimum, we should move it into the common.services package. But perhaps we can go further and just merge it function directly into ServiceConfiguration?

Comment (rotty3000) [reply org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceConfiguration.java]: Done!

File: org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceConfiguration.java [hunk 2]
```diff
@@ -125,36 +178,75 @@ public S getService(ServiceReference<S> reference) {
...
 
 	@Override
 	public S waitForService(long timeout) throws InterruptedException {
-		return getTracker().waitForService(timeout);
+		return tracker.waitForService(timeout);
 	}
 
-	private ServiceTracker<S, S> getTracker() {
-		return trackServices.tracker();
+	private class InnerCustomizer implements ServiceTrackerCustomizer<S, S> {
```
Comment (bjhargrave) [inline org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceConfiguration.java:209]: Should this be a static inner class? I don't think it depends upon anything in the outer class.

Comment (rotty3000) [reply org.osgi.test.common/src/main/java/org/osgi/test/common/service/ServiceConfiguration.java]: fixed

**Pull request merged**